gapic/src/main/com/google/gapid/models/Profile.java

```diff
@@ -119,10 +129,32 @@ public int hashCode() {
 
   public static class Data extends DeviceDependentModel.Data {
     public final Service.ProfilingData profile;
+    private final Map<Integer, Long> durationByGroup;
+    private final List<Service.ProfilingData.GpuSlices.Group> groups;
 
     public Data(Path.Device device, Service.ProfilingData profile) {
       super(device);
       this.profile = profile;
+      this.durationByGroup = aggregateSliceTimeByGroup(profile);
+      this.groups = getSortedGroups(profile, durationByGroup.keySet());
+    }
+
+    private static Map<Integer, Long> aggregateSliceTimeByGroup(Service.ProfilingData profile) {
+      Set<Integer> groups = profile.getSlices().getGroupsList().stream()
+          .map(Service.ProfilingData.GpuSlices.Group::getId)
+          .collect(toSet());
+      return profile.getSlices().getSlicesList().stream()
```

Comment:
On Mali, this may add up to as much as twice the actual wall clock time due to the two parallel queues. Do we just not care for now?